activestorage/lib/active_storage/engine.rb

```diff
@@ -157,5 +158,17 @@ class Engine < Rails::Engine # :nodoc:
         ActiveStorage::FixtureSet.file_fixture_path = ActiveSupport::TestCase.file_fixture_path
       end
     end
+
+    initializer "active_storage.old_signed_id_fallback" do
+      config.after_initialize do |app|
+        if ActiveStorage.old_signed_id_fallback
+          ActiveSupport::Deprecation.warn(<<-MSG.squish)
```

Comment:
Typically I think we show deprecation warnings on the behaviour we are switching on. That would be in `lib/active_storage/attached/changes/create_one.rb `. If we find a record with the old signing purpose, we should log a deprecation warning.

Reply:
Hey thanks for the review! I didn't quite follow this one, are you suggesting that we fire a warning on every blob we find with the old signature or just not fire at all?

Reply:
I mean instead of logging a deprecation here we should do it in `lib/active_storage/attached/changes/create_one.rb` instead. Something like this:

```
if ActiveStorage.old_signed_id_fallback
  if (blob = ActiveStorage::Blob.find_signed(attachable, record: record, purpose: :blob_id)
    # warn here
    blob
  else
    ActiveStorage::Blob.find_signed!(attachable, record: record, purpose: :direct_upload)
  end
else
  ActiveStorage::Blob.find_signed!(attachable, record: record, purpose: :direct_upload)
end
```

Reply:
Just in case, when addressing this comment we decided to always find with `purpose: :direct_upload` first and if not found then for `purpose: :blob_id` to prevent an extra query since in normal cases all new signatures will be using `purpose: :direct_upload`